scripts/o3de/o3de/manifest.py

```diff
@@ -563,29 +518,38 @@ def get_restricted_json_data(restricted_name: str = None, restricted_path: str o
     if restricted_name and not restricted_path:
         restricted_path = get_registered(restricted_name=restricted_name, project_path=project_path)
 
-    if not restricted_path:
-        logger.error(f'Restricted Path {restricted_path} has not been registered.')
+    return get_json_data('restricted', restricted_path, validation.valid_o3de_restricted_json)
+
+def get_repo_json_data(repo_uri: str = None) -> dict or None:
+    if not repo_uri:
+        logger.error('Must specify a Repo Uri.')
         return None
 
-    restricted_path = pathlib.Path(restricted_path).resolve()
-    restricted_json = restricted_path / 'restricted.json'
-    if not restricted_json.is_file():
-        logger.error(f'Restricted json {restricted_json} is not present.')
+    repo_json = get_repo_path(repo_uri=repo_uri)
+
+    if not repo_json.is_file():
+        logger.error(f'Repo json {repo_json} is not present.')
         return None
-    if not validation.valid_o3de_restricted_json(restricted_json):
-        logger.error(f'Restricted json {restricted_json} is not valid.')
+    if not validation.valid_o3de_repo_json(repo_json):
+        logger.error(f'Repo json {repo_json} is not valid.')
         return None
 
-    with restricted_json.open('r') as f:
+    with repo_json.open('r') as f:
         try:
-            restricted_json_data = json.load(f)
+            repo_json_data = json.load(f)
         except json.JSONDecodeError as e:
-            logger.warn(f'{restricted_json} failed to load: {str(e)}')
+            logger.warn(f'{repo_json} failed to load: {str(e)}')
         else:
-            return restricted_json_data
+            return repo_json_data
```

Comment:
Any reason this isn't calling the new `get_json_data` method?

Reply:
The functionality is a bit different but I can break get_json_data apart into 2 different functions and only call the part that is common among them all in get_repo_json_data